scripts/module_hash.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+"""
+Generate hash based on vetted versions of PETSc and libMesh and the
+configs used to build them.
+"""
+
+import os
+import sys
+import argparse
+import hashlib
+import subprocess
+
+# Vetted libraries used in the MOOSE repo. Index position matters (items on right
+# depend on items to the left)
+entities = ['petsc', 'libmesh']
+
+# MOOSE_DIR
+root_path = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(__file__)), '..'))
+
+def parse_args():
+    """ parse arguments """
+    parser = argparse.ArgumentParser(description='Supplies a hash for a given library'
+                                     ' based on MOOSE HEAD.')
+    parser.add_argument('library', nargs="+", metavar="library", choices=entities,
+                        help=f'choose from: {", ".join(entities)}')
+    return parser.parse_args()
+
+def build_list(args):
+    """ return list of files associated with supplied library """
+    _ltmp = []
+    for entity in entities:
+        _ltmp.extend([os.path.join(root_path, 'conda', f'{entity}', 'meta.yaml'),
+                      os.path.join(root_path, 'scripts', f'configure_{entity}.sh'),
+                      os.path.join(root_path, 'scripts', f'update_and_rebuild_{entity}.sh'),
+                      os.path.join(root_path, entity)])
+        if args.library[0] == entity:
+            break
+    return _ltmp
+
+def git_hash(submodule):
+    """ use git ls-tree HEAD PATH, to supply submodule hash """
+    with subprocess.Popen(['git',
+                           'ls-tree',
+                           'HEAD',
+                           os.path.join(root_path, submodule)],
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE,
+                          text=True,
+                          shell=False) as proc:
+        proc.wait()
+        if proc.poll() == 0:
+            return proc.communicate(0)[0].split()[2]
+        print(proc.communicate(1))
+        sys.exit(1)
```

Comment:
```suggestion
    out = subprocess.check_output(['git', 'ls-tree', 'HEAD', os.path.join(root_path, submodule)]).decode('utf-8')
    return out.split()[2]
```

Reply:
Do we not want to handle an error?

Reply:
If I remember right, check output prints stderr to screen and throws an exception on a nonzero return code